### PR TITLE
bpo-24758: Improve the error msg for unittest.mock.Mock()'s unsafe mode

### PR DESCRIPTION
**Pull request by ZackerySpytz**
https://bugs.python.org/issue24758

### CODE REVIEW

File: Lib/unittest/mock.py [hunk 1]
```diff
@@ -595,7 +595,7 @@ def __getattr__(self, name):
             raise AttributeError(name)
         if not self._mock_unsafe:
             if name.startswith(('assert', 'assret')):
-                raise AttributeError(name)
+                raise AttributeError("Attributes cannot start with 'assert'")
```
Comment (tirkarthi) [inline Lib/unittest/mock.py]: `startswith` checks both "assert" and "assret" so hardcoding "assert" might lead to confusing error message when the attribute starts with assret instead. Perhaps change this to "Attributes cannot start with 'assert' or 'assret'" ?

```
./python.exe
Python 3.8.0a3+ (heads/master:55d035113d, Apr 28 2019, 01:37:48)
[Clang 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from unittest import mock
>>> mock.Mock().assret_mock_call
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/karthikeyansingaravelan/stuff/python/cpython/Lib/unittest/mock.py", line 598, in __getattr__
    raise AttributeError("Attributes cannot start with 'assert'")
AttributeError: Attributes cannot start with 'assert'
```

**Review by kushaldas — CHANGES_REQUESTED** [review]
`"Attributes cannot start with 'assert' or 'assret'"` is a better error message in this. Also, please check for the right error message in `test_mock_unsafe` as we are checking the assertion there.

**Comment by ZackerySpytz** [comment]
Thank you, @kushaldas.

I have made the requested changes; please review again.

**Review by tirkarthi — APPROVED** [review]
LGTM. The attribute can start with assert or assret in unsafe mode but adding it to error message looks intrusive to me and this seems like a good improvement.

Thank you for the PR.

**Review by kushaldas — APPROVED** [review]
This looks good now, thank you for the PR. Approved.

**Pull request merged**